src/main/java/org/truffleruby/language/LazyRubyNode.java

```diff
@@ -62,4 +61,21 @@ public RubyNode resolve() {
         });
     }
 
+    private RubyNode getOrCreateMasterResolution() {
+        synchronized (resolver) {
+            RubyNode masterResolution = resolutionMaster[0];
+            if (masterResolution == null) {
```

Comment:
Would Double-checked locking be useful here?
cc: @eregon

Reply:
Yes, we can do that.

Reply:
No. It's behind a deoptimization and only hit once before first execution.
I am sorry but this a premature optimization which only makes the code more complex here.